### PR TITLE
Api validations

### PR DESCRIPTION
**Pull request by Sakaa4Short**

### CODE REVIEW

File: spec/models/cat_spec.rb [hunk 1]
```diff
@@ -1,5 +1,14 @@
 require 'rails_helper'
 
 RSpec.describe Cat, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
-end
+  it "should validate name" do
+    cat = Cat.create(age: 2, enjoys: 'Walks in the park', image: 'https://images.unsplash.com/photo-1529778873920-4da4926a72c2?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1036&q=80')
+    expect(cat.errors[:name]).to_not be_empty
+    # expect(cat.errors[:age]).to_not be_empty
+    # expect(cat.errors[:image]).to_not be_empty
```
Comment (sjproctor) [inline spec/models/cat_spec.rb]: What are line 7 and 8 doing?

File: spec/requests/cats_spec.rb [hunk 2]
```diff
@@ -44,4 +44,41 @@
       expect(cat.name).to eq 'Buster'
     end
   end
+
+  # describe "PATCH/ update" do
```
Comment (sjproctor) [inline spec/requests/cats_spec.rb]: What is this test doing?

**Review by sjproctor — COMMENTED** [review]
Your branch says this is API endpoints but your code doesn't communicate that. Please update your code to match the expectation of this card.

File: spec/models/cat_spec.rb (hunk 1 above)
Comment (Sakaa4Short) [reply spec/models/cat_spec.rb]: Removed Lined Out Code

File: spec/requests/cats_spec.rb (hunk 2 above)
Comment (Sakaa4Short) [reply spec/requests/cats_spec.rb]: Removed Inactive Tests

File: spec/models/cat_spec.rb [hunk 3]
```diff
@@ -1,5 +1,12 @@
 require 'rails_helper'
 
 RSpec.describe Cat, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
-end
+  it "should validate name" do
+    cat = Cat.create(age: 2, enjoys: 'Walks in the park', image: 'https://images.unsplash.com/photo-1529778873920-4da4926a72c2?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1036&q=80')
+    expect(cat.errors[:name]).to_not be_empty
+  end
```
Comment (sjproctor) [inline spec/models/cat_spec.rb:7]: You need to also include validations for presence of age, enjoys, and image.

File: spec/requests/cats_spec.rb [hunk 4]
```diff
@@ -44,4 +44,22 @@
...
+    }
+    # Send the request to the  server
+    post '/cats', params: cat_params
+    # expect an error if the cat_params does not have a name
+    expect(response.status).to eq 422
+    # Convert the JSON response into a Ruby Hash
+    json = JSON.parse(response.body)
+    # Errors are returned as an array because there could be more than one, if there are more than one validation failures on an attribute.
+    expect(json['name']).to include "can't be blank"
+  end
```
Comment (sjproctor) [inline spec/requests/cats_spec.rb:64]: We should also get a 422 for age, enjoys, and image.

**Review by sjproctor — COMMENTED** [review]
Name looks good. But we need the same for age, enjoys, and image.

**Review by sjproctor — APPROVED** [review]

**Pull request merged**